### ISSUE TITLE
add_contours does not work with negative maps

### ISSUE THREAD

**Issue by lrq3000** [nilearn/nilearn#1596]
Hey there!

`add_contours()` does not seem to work with negative maps, only positive ones. It would be nice to make the function works with both high positive and high negative values.

Currently, the workaround is to compute the absolute of all voxels intensities first:

```python
import nibabel as nib
import numpy as np
im_abs = nib.Nifti1Image(np.absolute(im.get_data()), affine=im.affine)
fig.add_contours(im_abs)
```

Thanks!

**Comment by KamalakerDadi** [comment]
I agree this minimum feature ```add_contours``` is fragile with negative maps. But, I would like you to try the approach told in this [example](http://nilearn.github.io/auto_examples/01_plotting/plot_demo_glass_brain_extensive.html#sphx-glr-auto-examples-01-plotting-plot-demo-glass-brain-extensive-py) in **Section 8.2.16.5. Display contour projections in both hemispheres**. How to plot negative maps with add_contours. Please let us know if you have tried this ?

**Comment by lrq3000** [comment]
@KamalakerDadi My bad, I did not see this example, indeed it works with simply supplying `levels=[-0.1, 0.1], colors=['b', 'r']`, and this works even if we supply the positive and negative images separately using two different calls to `add_contours()` (as I am doing).

BTW would someone know why using `add_contours` might change the size of the visualization?

With contours (notice the last set of images on the right):
![not-same-size2](https://user-images.githubusercontent.com/1118942/35410017-4231d6d8-0214-11e8-8750-ad571adf083c.png)

Without contours, no size issue:
![same-size-no-contours](https://user-images.githubusercontent.com/1118942/35410073-6d048b08-0214-11e8-820d-ae5c6cfa4045.png)

I checked and it's not because of some parameters (will happen even if using a levels value that prevent the display of any contour) nor subplotting.

**Comment by lrq3000** [comment]
Ok I fixed it, maybe not elegantly, but it works: in this file: https://github.com/nilearn/nilearn/blob/2520333761ddee468cfb948d5ad12525d9d186c3/nilearn/plotting/displays.py#L703

Replace:

```python
im = display_ax.draw_2d(data_2d, data_bounds, bounding_box,
                                        type=type, **kwargs)
```

by:

```python
im = display_ax.draw_2d(data_2d, data_bounds, bounding_box if auto_resize else [(0,0),(0,0),(0,0)],
                                        type=type, **kwargs)
```

Then add an addition argument `auto_resize=True` to `_map_show()` and now it's possible to do:

```python
fig.add_contours(im, levels=[-0.1, 0.1], colors=['b', 'r'], linewidths=1., auto_resize=False)
```

Which correctly displays the image in original size and with contours:
![same-size-with-contours](https://user-images.githubusercontent.com/1118942/35410950-4707f626-0217-11e8-95db-b29b98fadb4f.png)

**Comment by bthirion** [comment]
Thx for finding the fix.
I think it could be included in the library, if you want to issue a PR.

**Comment by lrq3000** [comment]
@bthirion If you think this fix is correct then I'll make a PR :-) (although with some delay, I have a very urgent deadline atm!)

**Comment by lrq3000** [comment]
Well, I did the patch, but now the example to plot negative contours does not work anymore (I messed up a bit with my packages to install niwidgets, I had to update everything).

So to recap: plotting positive+negative contours works as per the documentation, but plotting separately one map as positive contour and a second map as negative contour, thus by using two calls to `add_contours()`, will not work anymore (although it worked before? o_O).

I have absolutely no idea what is the root cause...
